sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -1059,8 +1059,8 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog with QueryErrorsB
       if (!canHostOuter(p) && p.expressions.exists(containsOuter)) {
         p.failAnalysis(
           errorClass =
-            "UNSUPPORTED_SUBQUERY_EXPRESSION_CATEGORY.UNSUPPORTED_CORRELATED_REFERENCE",
-          messageParameters = Map("treeNode" -> planToString(p)))
+            "UNSUPPORTED_SUBQUERY_EXPRESSION_CATEGORY.CORRELATED_REFERENCE",
+          messageParameters = Map("sqlExprs" -> p.expressions.map(_.sql).mkString(",")))
```

Comment:
Could you show only expressions for which `containsOuter()` returns `true`.

Reply:
Maybe it's a dumb question, but it's already checked in `if` clause above ??

```
if (!canHostOuter(p) && p.expressions.exists(containsOuter)) {
```

Is it different one?

Reply:
`exists()` returns `true` if at least ONE element satisfies the requirement but  you show ALL.

Reply:
Oh, I got it. Let me address it.

Thanks!!